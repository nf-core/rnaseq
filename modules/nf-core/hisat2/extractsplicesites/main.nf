process HISAT2_EXTRACTSPLICESITES {
    tag "$gtf"
    label 'process_medium'

    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/hisat2:2.2.1--h1b792b2_3' :
        'biocontainers/hisat2:2.2.1--h1b792b2_3' }"

    input:
    tuple val(meta), path(gtf)

    output:
    tuple val(meta), path("*.splice_sites.txt"), emit: txt
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def VERSION = '2.2.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    hisat2_extract_splice_sites.py $gtf > ${gtf.baseName}.splice_sites.txt
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hisat2: $VERSION
    END_VERSIONS
    """
}
