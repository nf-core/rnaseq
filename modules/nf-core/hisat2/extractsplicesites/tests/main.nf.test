nextflow_process {

    name "Test Process HISAT2_EXTRACTSPLICESITES"
    script "../main.nf"
    process "HISAT2_EXTRACTSPLICESITES"
    tag "modules"
    tag "modules_nfcore"
    tag "hisat2"
    tag "hisat2/extractsplicesites"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [id:'genome'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.gtf', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path("${process.out.txt[0][1]}").exists() },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
