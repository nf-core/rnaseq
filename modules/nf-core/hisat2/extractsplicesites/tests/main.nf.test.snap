{
    "Should run without failures": {
        "content": [
            [
                "versions.yml:md5,eeea7231fe197810659b8bad4133aff2"
            ]
        ],
        "timestamp": "2024-01-18T20:56:30.71763"
    }
}